src/herder/test/HerderTests.cpp

```diff
@@ -3534,6 +3534,91 @@ TEST_CASE("soroban txs each parameter surge priced")
+{
+    auto networkID = sha256(getTestConfig().NETWORK_PASSPHRASE);
+
+    auto simulation =
+        Topologies::core(4, 1, Simulation::OVER_LOOPBACK, networkID, [](int i) {
+            auto cfg = getTestConfig(i, Config::TESTDB_ON_DISK_SQLITE);
+            cfg.TESTING_UPGRADE_MAX_TX_SET_SIZE = 100;
+            cfg.TESTING_UPGRADE_LEDGER_PROTOCOL_VERSION =
+                static_cast<uint32_t>(SOROBAN_PROTOCOL_VERSION) - 1;
+            return cfg;
+        });
+
+    simulation->startAllNodes();
+    auto nodes = simulation->getNodes();
+    uint32_t numAccounts = 100;
+    auto& loadGen = nodes[0]->getLoadGenerator();
+
+    // Generate some accounts
+    auto& loadGenDone =
+        nodes[0]->getMetrics().NewMeter({"loadgen", "run", "complete"}, "run");
+    auto currLoadGenCount = loadGenDone.count();
+    loadGen.generateLoad(
+        GeneratedLoadConfig::createAccountsLoad(numAccounts, 1));
+    simulation->crankUntil(
+        [&]() { return loadGenDone.count() > currLoadGenCount; },
+        10 * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
+
+    // Ensure more transactions get in the ledger post upgrade
+    ConfigUpgradeSetFrameConstPtr res;
+    Upgrades::UpgradeParameters scheduledUpgrades;
+    scheduledUpgrades.mUpgradeTime =
+        VirtualClock::from_time_t(nodes[0]
+                                      ->getLedgerManager()
+                                      .getLastClosedLedgerHeader()
+                                      .header.scpValue.closeTime +
+                                  15);
+    scheduledUpgrades.mProtocolVersion =
+        static_cast<uint32_t>(SOROBAN_PROTOCOL_VERSION);
+    for (auto const& app : nodes)
+    {
+        app->getHerder().setUpgrades(scheduledUpgrades);
+    }
+
+    auto& secondLoadGen = nodes[1]->getLoadGenerator();
+    auto& secondLoadGenDone =
+        nodes[1]->getMetrics().NewMeter({"loadgen", "run", "complete"}, "run");
+    currLoadGenCount = loadGenDone.count();
+    auto secondLoadGenCount = secondLoadGenDone.count();
+
+    // Generate classic txs from another node (with offset to prevent
+    // overlapping accounts)
+    secondLoadGen.generateLoad(GeneratedLoadConfig::txLoad(LoadGenMode::PAY, 50,
+                                                           /* nTxs */ 100, 2,
+                                                           /* offset */ 50));
+
+    // Crank a bit and verify that upgrade went through
+    simulation->crankForAtLeast(4 * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
+    REQUIRE(nodes[0]
+                ->getLedgerManager()
+                .getLastClosedLedgerHeader()
+                .header.ledgerVersion ==
+            static_cast<uint32_t>(SOROBAN_PROTOCOL_VERSION));
+
+    // Now generate Soroban txs
+    auto sorobanConfig =
+        GeneratedLoadConfig::txLoad(LoadGenMode::SOROBAN, 50,
+                                    /* nTxs */ 15, 1, /* offset */ 0);
+    sorobanConfig.skipLowFeeTxs = true;
```

Comment:
Do we need this because we don't have even loadgen spacing for soroban yet? If so, we need to at least check that *some* txs succeeded, otherwise it's possible to pass the test while not accepting a single soroban tx.

Reply:
It's because when we upgrade to protocol 20, our Soroban limits are too conservative (even 1 tx/s rate is too high). Technically, we could do another upgrade and bump ledger-wide limits, but for the purpose of this test I just wanted to make sure the queue is initialized and nodes are able to include Soroban txs. Added a check that _some_ txs went in based on your suggestions, so I think that should be sufficient.